Move functorch tests from functorch/test/* to test/functorch/*

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #86623

This is the first step described in
https://github.com/pytorch/pytorch/issues/86618 . test/functorch/* is
the final location for these tests.

Test Plan:
- Check that the functorch shards in CI are still running tests.

zou3519 committed:
Move functorch tests from functorch/test/* to test/functorch/*

This is the first step described in
https://github.com/pytorch/pytorch/issues/86618 . test/functorch/* is
the final location for these tests.

Test Plan:
- Check that the functorch shards in CI are still running tests.

[ghstack-poisoned]

samdow commented on test/run_test.py:
```diff
@@ -66,8 +66,7 @@ def skip_test_p(name: str) -> bool:
-    # This supports symlinks, so we can link domain library tests like functorch
-    # to PyTorch test directory
+    # This supports symlinks, so we can link domain library tests to PyTorch test directory
     all_py_files = [pathlib.Path(p) for p in glob.glob(f"{cwd}/**/test_*.py", recursive=True)]
```
cc @huydhn who did this change originally to get us sharding

Just want to check if you think we should keep it like this to support future symlinks or change back anything else originally done in #85321

huydhn replied:
I would prefer to keep the symlink logic here because it's more flexible than the old logic in itself.  For functorch, this PR has already done exactly what needs to be done 1) removing the symlink to `functorch/test` and 2) moving all the tests to `test/functorch`.

huydhn reviewed (approved):
LGTM! There are only some minor lint error about setting the owner of the new tests.  Also I have already attach the `skip-pr-sanity-checks` label to the PR, so that check should be bypassed now

zou3519 committed:
Update on "Move functorch tests from functorch/test/* to test/functorch/*"


This is the first step described in
https://github.com/pytorch/pytorch/issues/86618 . test/functorch/* is
the final location for these tests.

Test Plan:
- Check that the functorch shards in CI are still running tests.

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge

zou3519 commented:
@pytorchbot help
